### PR TITLE
[CI] Fix `kitchen_deploy` stage tests by avoiding race conditions

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

This PR adds `resource_group` attribute to the `deploy_deb` and `deploy_yum` jobs to avoid writing simultaneously to the repo metadata file.

### Motivation

We sometimes have issues in kitchen tests because metadata files aren’t correct due 

For instance, [this deploy job](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/191198298) published the arm64 Agent 7 deb package for pipeline 11035730, but the metadata file reads:
```
Codename: pipeline-11035730-a7
Date: Wed, 09 Nov 2022 13:54:18 UTC
Architectures: amd64 i386 armhf x86_64
Components: 7
```

because of two simultaneous writes to the repo metadata file, causing failures in kitchen tests: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/191285322.

### Describe how to test/QA your changes

Run the pipeline and check that the `Kitchen_deploy` stage tests are not failing. Concerned tests are:
- `deploy_deb_testing-a6_x64` and `deploy_deb_testing-a6_arm64`.
- `deploy_deb_testing-a7_x64` and `deploy_deb_testing-a7_arm64`.
- `deploy_rpm_testing-a6_x64` and `deploy_rpm_testing-a6_arm64`.
- `deploy_rpm_testing-a7_x64` and `deploy_rpm_testing-a7_arm64`.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: .gitlab/kitchen_deploy.yml [hunk 1]
```diff
@@ -153,6 +170,7 @@ deploy_rpm_testing-a6_arm64:
   stage: kitchen_deploy
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-builders/gitlab_agent_deploy:$DATADOG_AGENT_BUILDERS
   tags: ["runner:main"]
+  <<: *deploy_deb_resource_group-a7
```
Comment (bkabrda) [inline .gitlab/kitchen_deploy.yml:173]:
```suggestion
  <<: *deploy_rpm_resource_group-a7
```

File: .gitlab/kitchen_deploy.yml [hunk 2]
```diff
@@ -122,6 +138,7 @@ deploy_deb_testing-a7_arm64:
   stage: kitchen_deploy
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-builders/gitlab_agent_deploy:$DATADOG_AGENT_BUILDERS
   tags: ["runner:main"]
+  <<: *deploy_deb_resource_group-a6
```
Comment (bkabrda) [inline .gitlab/kitchen_deploy.yml:141]:
```suggestion
  <<: *deploy_rpm_resource_group-a6
```

**Review by bkabrda — CHANGES_REQUESTED** [review]

File: .gitlab/kitchen_deploy.yml [hunk 3]
```diff
@@ -36,10 +36,25 @@
...
+  resource_group: deploy_deb_a6
+
+.deploy_deb_resource_group-a7: &deploy_deb_resource_group-a7
+  resource_group: deploy_deb_a7
+
+.deploy_rpm_resource_group-a6: &deploy_rpm_resource_group-a6
+  resource_group: deploy_rpm_a6
+
+.deploy_rpm_resource_group-a7: &deploy_rpm_resource_group-a7
+  resource_group: deploy_rpm_a7
```
Comment (KSerrania) [inline .gitlab/kitchen_deploy.yml:50]: I believe the rpm parts aren't needed.

The reason why the deb parts are needed are due to the deb repository structure:
```
apttesting.datad0g.com/
  dists/
    pipeline-<pipeline id>-<agent version>/
      Release <- Shared metadata, written to by both amd64 and arm64 deploy jobs
      Release.gpg
      <agent version>/
        binary-amd64/
          Packages <- Specific metadata, written to by the amd64 deploy job
        binary-arm64/
          Packages <- Specific metadata, written to by the arm64 deploy job
```

In the rpm repo, we don't have such shared metadata:
```
yumtesting.datad0g.com/
  testing/
    pipeline-<pipeline id>-<agent version>/
      <agent version>/
        aarch64/
          repodata/ <- Specific metadata, written to by the amd64 deploy job
        amd64/
          repodata/ <- Specific metadata, written to by the arm64 deploy job
```

File: .gitlab/kitchen_deploy.yml (hunk 2 above)
Comment (KSerrania) [inline .gitlab/kitchen_deploy.yml:141]: Slavek's comment that I deleted by mistake:

```suggestion
  <<: *deploy_rpm_resource_group-a6
```

File: .gitlab/kitchen_deploy.yml (hunk 3 above)
Comment (amenasria) [reply .gitlab/kitchen_deploy.yml]: Thanks for the detailed explanation ! Fixed in [0126fa2](https://github.com/DataDog/datadog-agent/pull/14398/commits/0126fa27e28b95056d4925a838230c3a634e853a)

**Review by KSerrania — APPROVED** [review]

**Pull request merged**